File: ios/PlayerViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(PlayerViewManager, RCTViewManager)

// Initial Props
RCT_EXPORT_VIEW_PROPERTY(ServerUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(PartnerId, NSString)
RCT_EXPORT_VIEW_PROPERTY(UiConfId, NSString)

RCT_EXTERN_METHOD(
  play:(nonnull NSNumber *)node
)

RCT_EXTERN_METHOD(
  pause:(nonnull NSNumber *)node
)

RCT_EXTERN_METHOD(
  toggleFullScreen:(nonnull NSNumber *)node
)

@end

File: ios/PlayerViewManager.swift
import Foundation

@objc(PlayerViewManager)
class PlayerViewManager: RCTViewManager {
  
  override func view() -> UIView! {
    return PlayerView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func pause(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! PlayerView
      
      component.pause()
    }
  }
  
  @objc func play(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! PlayerView
      
      component.play()
    }
  }
  
  @objc func toggleFullScreen(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(
        forReactTag: node
      ) as! PlayerView
      
      component.toggleFullScreen()
    }
  }
}
